magpy/stream.py

```diff
@@ -4109,6 +4109,14 @@ def fit(self, keys, **kwargs):
                 f_fit = self.harmfit(nt, val, fitdegree)
                 # Don't use resampled list for harmonic time series
                 x = nt
+            elif len(val)>1 and fitfunc == 'linear least-squares':
+                logger.debug('Selected linear least-squares fit')
+                A = np.vstack([nt, np.ones(len(nt))]).T
+                m, c, = np.linalg.lstsq(A, val)[0]
+                f_fit = m * x + c
+            elif fitfunc == 'none':
+                logger.debug('Selected no fit')
+                return
             elif len(val)<=1:
```

Comment:
I recommend moving this case to top of the if statements, so the rest do not need to include `len(val) > 1`...